torch/distributed/_functional_collectives_impl.py

```diff
@@ -0,0 +1,246 @@
+import warnings
+import weakref
+import torch
+import torch.distributed as dist
+import torch.distributed.distributed_c10d as c10d
+from typing import List, cast
+
+"""
+Moved eager kernel implementations to a separate file partly for readability and partly as it is currently
+easier in dynamo to set tracing policy on a file-by-file level.
```

Comment:
This is interesting, are there some guidelines of what should go on this file and what should go in _functional_collectives.py?